tests/testutils/iputils.go

```diff
@@ -88,3 +92,87 @@ func GetGWv4(iface string) (string, error) {
 func GetGWv6(iface string) (string, error) {
 	return GetGW(iface, netlink.FAMILY_V4)
 }
+
+func GetNonLoIfaceWithAddrs() (iface net.Interface, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return iface, err
+	}
+
+	for _, i := range ifaces {
+		if i.Flags&net.FlagLoopback == 0 {
+			ifaceNameLower := strings.ToLower(i.Name)
+			// Don't use rkt's interfaces
+			if strings.HasSuffix(ifaceNameLower, "cni") ||
+				strings.HasSuffix(ifaceNameLower, "veth") {
+				continue
+			}
+			addrs, err := i.Addrs()
+			if err != nil {
+				return iface, fmt.Errorf("Cannot get IPV4 address for interface %v: %v", i.Name, err)
+			}
+			if len(addrs) > 0 {
+				iface = i
+				break
+			}
+		}
+	}
+	return iface, err
+}
+
+func GetNonLoIfaceIPv4() (string, error) {
+	iface, err := GetNonLoIfaceWithAddrs()
+	if err != nil {
+		return "", fmt.Errorf("Error while getting non-lo host interface: %v\n", err)
+	}
+	if iface.Name == "" {
+		return "", nil
+	}
+
+	ifaceIPsv4, err := GetIPsv4(iface.Name)
+	if err != nil {
+		return "", fmt.Errorf("Cannot get IPV4 address for interface %v: %v", iface.Name, err)
+	}
+	if len(ifaceIPsv4) == 0 {
+		return "", nil
+	}
+	return ifaceIPsv4[0], nil
+}
+
+func CheckTcp4Port(port int) (bool, error) {
+	tcpFile, err := os.Open("/proc/net/tcp")
+	if err != nil {
+		return false, err
+	}
+	defer tcpFile.Close()
+
+	scanner := bufio.NewScanner(tcpFile)
+	for scanner.Scan() {
+		re := regexp.MustCompile(`:([A-Z0-9]+) `)
```

Comment:
Small nit. I know this is not performance critical but I would still move this out of the loop.